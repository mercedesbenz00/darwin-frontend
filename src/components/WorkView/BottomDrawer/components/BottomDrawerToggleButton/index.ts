export { default as BottomDrawerToggleButton } from './BottomDrawerToggleButton.vue'
