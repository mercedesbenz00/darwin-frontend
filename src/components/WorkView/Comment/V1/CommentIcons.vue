<template>
  <div class="comment-icons">
    <comment-thread-icon
      v-for="commentThread in filteredCommentThreads"
      :key="commentThread.id"
      :comment-thread="commentThread"
      :view="view"
    />
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'
import { State } from 'vuex-class'

import { CommentThread, View } from '@/engine/models'
import { RootState } from '@/store/types'

import CommentThreadIcon from './CommentThreadIcon.vue'

@Component({ name: 'comment-icons', components: { CommentThreadIcon } })
export default class CommentIcons extends Vue {
  @State((state: RootState) => state.comment.commentThreads.filter(t => !t.resolved))
  commentThreads!: CommentThread[]

  @Prop({ required: true, type: Object as () => View })
  view!: View

  /**
   * Load frame-specific comments for videos
   */
  get filteredCommentThreads () {
    const { loadedVideo } = this.view
    if (!loadedVideo) {
      return this.commentThreads
    }

    return this.commentThreads.filter(commentThread =>
      commentThread.frameIndex === this.view.currentFrameIndex
    )
  }
}
</script>

<style lang="scss" scoped>
.comment-icons {
  position: absolute;
  top: 0;
  left: 0;
}
</style>
