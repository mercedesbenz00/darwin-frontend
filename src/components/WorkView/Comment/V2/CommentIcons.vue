<template>
  <div class="comment-icons">
    <comment-thread-icon
      v-for="thread in threads"
      :key="thread.id"
      :thread-id="thread.id"
      :view-id="viewId"
    />
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

import { useCurrentFrameThreads, useView } from '@/composables/useEditorV2'

import CommentThreadIcon from './CommentThreadIcon.vue'

/**
 * Renders a list of icon components, one per comment thread, on top of the canvas.
 *
 * Does not deal with positioning. Each individual icon is in charge of their own.
 */
export default defineComponent({
  name: 'CommentIcons',
  components: { CommentThreadIcon },
  props: {
    viewId: { type: String, required: true }
  },
  setup (props) {
    const view = useView(props.viewId)
    const threads = useCurrentFrameThreads(view)

    return { threads }
  }
})
</script>

<style lang="scss" scoped>
.comment-icons {
  position: absolute;
  top: 0;
  left: 0;
}
</style>
