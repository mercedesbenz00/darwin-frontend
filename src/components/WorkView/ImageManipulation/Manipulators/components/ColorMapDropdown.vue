<template>
  <div class="color-map-dropdown">
    <div class="dropdown">
      <Select2
        v-model="sel"
        :options="options"
        :settings="settings"
        @select="onSelect($event)"
      />
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'

import Select2 from '@/components/Common/Select2.vue'
import { ColorMap } from '@/engineCommon/imageManipulation'

export type ColorMapDropdownOption = {
  id: ColorMap
  icon?: string
  text: string
}

@Component({
  name: 'color-map-dropdown',
  components: { Select2 }
})
export default class ColorMapDropdown extends Vue {
  @Prop({ required: true })
  value!: ColorMap

  get options (): ColorMapDropdownOption[] {
    return [
      { id: 'default', text: 'Raw (default)' },
      { id: 'bone', text: 'Bone', icon: '/static/imgs/color-maps/bone.svg' },
      { id: 'jet', text: 'Jet', icon: '/static/imgs/color-maps/jet.svg' },
      { id: 'hot', text: 'Hot', icon: '/static/imgs/color-maps/hot.svg' },
      { id: 'seismic', text: 'Seismic', icon: '/static/imgs/color-maps/seismic.svg' },
      { id: 'paired', text: 'Paired', icon: '/static/imgs/color-maps/paired.svg' }
    ]
  }

  get sel () {
    return this.value
  }

  set sel (val) {
    this.$emit('input', val)
    this.$emit('change', val)
  }

  get settings () {
    return {
      minimumResultsForSearch: -1,
      templateResult: this.formatTemplate.bind(this),
      templateSelection: this.formatTemplate.bind(this),
      theme: 'color-map',
      width: '100%'
    }
  }

  formatTemplate (state: ColorMapDropdownOption) {
    let iconHtml = ''
    if (state.icon) {
      iconHtml = `<img src=${state.icon} class="select2-results__option__icon">`
    }
    // eslint-disable-next-line
    return $(`
    <div class="select2-results__option-cover">
      <span class="select2-results__option__text">${state.text}</span> ${iconHtml}
    </div>
    `)
  }

  onSelect (option: ColorMapDropdownOption) {
    this.sel = option.id
  }
}
</script>

<style lang="scss" scoped>
.color-map-dropdown {
  position: relative;
  margin: 0 -10px;
}

.dropdown {
  position: relative;
  width: 100%;
  height: 35px;
  cursor: pointer;
}
</style>

<!-- eslint-disable vue-scoped-css/enforce-style-type -->
<style lang="scss">
$border-height: 2px;
$height: 36px;

.select2-container.select2-container--color-map {
  margin: 0;
  position: relative;
}
  .select2-container.select2-container--color-map .select2-selection--single {
    height: $height;
    outline: none;
    border: none;
    background-color: transparent;
    border-radius: $border-radius-default;
    overflow: hidden;
  }
    .select2-container.select2-container--color-map .select2-selection--single .select2-selection__rendered {
      border: none;
      @include typography(md-1, default);
      color: $color90Black;
      background: $colorAliceBlue;
      box-shadow: inset 0px 2px 4px rgba(145, 169, 192, 0.3);
      padding: 9px; }
    .select2-container.select2-container--color-map .select2-selection--single .select2-selection__rendered .select2-results__option-cover {
      @include row;
      align-items: center;
    }
      .select2-container.select2-container--color-map .select2-selection--single .select2-selection__rendered .select2-results__option-cover .select2-results__option__text {
        flex: 1;
      }

      .select2-container.select2-container--color-map .select2-selection--single .select2-selection__rendered .select2-results__option-cover .select2-results__option__icon {
        width: 34px;
        margin-right: 25px;
      }
    .select2-container.select2-container--color-map.select2-container--focus .select2-selection--single .select2-selection__rendered {
      padding: 9px; }
.select2-container.select2-container--color-map .select2-dropdown {
  background: $colorWhite;
  border: none;
  overflow: hidden;
  z-index: 1051; }

.select2-container--color-map .select2-results__option {
  @include typography(md, default);
  background: $colorWhite;
  color: $colorSecondaryLight;
  user-select: none;
  -webkit-user-select: none; }
.select2-container--color-map .select2-results__option .select2-results__option-cover {
  margin: 8px 10px;
  @include row;
  align-items: center;
}
  .select2-container--color-map .select2-results__option .select2-results__option-cover .select2-results__option__text {
    flex: 1;
  }
  .select2-container--color-map .select2-results__option .select2-results__option-cover .select2-results__option__icon {
    width: 34px;
    margin-right: 5px;
  }
.select2-container--color-map .select2-search--dropdown {
  padding: 4px; }
  .select2-container--color-map .select2-search--dropdown .select2-search__field {
    padding: 4px;
    box-sizing: border-box; }
.select2-container--color-map .select2-selection--single {
  background-color: transparent;
  border: none;
  border-radius: $border-radius-default; }
  .select2-container--color-map .select2-selection--single .select2-selection__rendered {
    color: $colorSecondaryDark1;
    height: $height;
    line-height: normal; }
  .select2-container--color-map .select2-selection--single .select2-selection__clear {
    cursor: pointer;
    float: right;
    font-weight: bold; }
  .select2-container--color-map .select2-selection--single .select2-selection__placeholder {
    @include typography(md-1, default);
    color: $colorSecondaryLight1; }
  .select2-container--color-map .select2-selection--single .select2-selection__arrow {
    width: 14px;
    height: 26px;
    position: absolute;
    top: 0;
    right: 9px; }
    .select2-container--color-map .select2-selection--single .select2-selection__arrow b {
      position: absolute;
      border-left: 7px solid transparent !important;
      border-right: 7px solid transparent !important;
      border-top: 7px solid $colorPrimaryLight !important;
      border-bottom: none !important;
      left: auto;
      top: 55%;
      margin: 0;
      height: 0;
      width: 0; }
  .select2-container--color-map.select2-container--disabled .select2-selection--single .select2-selection__arrow {
    display: none;
  }
.select2-container--color-map.select2-container--open .select2-selection--single .select2-selection__rendered {
  padding: 9px; }
.select2-container--color-map.select2-container--disabled .select2-selection--single {
  background-color: inherit;
  cursor: default; }
  .select2-container--color-map.select2-container--disabled .select2-selection--single .select2-selection__clear {
    display: none; }
.select2-container--color-map.select2-container--open .select2-selection--single .select2-selection__arrow b {
  border: none;
  border-left: 7px solid transparent !important;
  border-right: 7px solid transparent !important;
  border-top: none !important;
  border-bottom: 7px solid $colorPrimaryLight !important; }
.select2-container--color-map .select2-search--dropdown .select2-search__field {
  box-shadow: 0px 10px 60px rgba(11, 36, 72, 0.1), 0px 1px 2px rgba(11, 36, 72, 0.05);
  @include typography(md, headlines);
  letter-spacing: 0.5px;
  color: $colorSecondaryDark1;
  border-radius: 5px;
  border: 1px solid $colorSecondaryLight1; }
.select2-container--color-map .select2-results > .select2-results__options {
  max-height: 200px;
  overflow-y: auto; }
.select2-container--color-map .select2-results__option--highlighted:nth-child(odd) {
  background: $colorGriteDark;
}
.select2-container--color-map .select2-results__option--highlighted:nth-child(odd)[aria-selected] {
  font-weight: bold;
  color: $colorSecondaryDark1;
}
.select2-container--color-map .select2-results__option--highlighted:nth-child(even) {
  background: $colorWhite;
}
.select2-container--color-map .select2-results__option--highlighted:nth-child(even)[aria-selected] {
  font-weight: bold;
  color: $colorSecondaryDark1;
}
.select2-container--color-map .select2-results__option[aria-selected=true]:nth-child(odd) {
  background: $colorGriteDark; }
.select2-container--color-map .select2-results__option[aria-selected=true]:nth-child(even) {
  background: $colorWhite; }
.select2-container--color-map .select2-results__option--highlighted[aria-selected] {
  background: $colorAliceBlue;
  font-weight: bold;
  color: $colorSecondaryDark1; }
.select2-container--color-map .select2-results__option--highlighted[aria-selected]:hover {
  background: $colorAliceBlue;
}
.select2-container--color-map.select2-container--open .select2-dropdown--above {
  box-shadow: 0px -5px 60px rgba(11, 36, 72, 0.2), 0px 1px 2px rgba(58, 78, 108, 0.05);
  border-bottom: none;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px; }
.select2-container--color-map.select2-container--open .select2-dropdown--below {
  box-shadow: 0px 5px 60px rgba(11, 36, 72, 0.2), 0px 1px 2px rgba(58, 78, 108, 0.05);
  border-top: none;
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px; }
</style>
