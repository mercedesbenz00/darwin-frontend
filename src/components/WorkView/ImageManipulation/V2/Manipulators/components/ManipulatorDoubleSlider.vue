<template>
  <vue-slider
    ref="slider"
    v-model="_value"
    v-bind="sliderOptions"
  >
    <template #dot="{ index }">
      <div
        class="slider__dot"
        :class="`slider__dot--${index}`"
      >
        <scrubber-icon class="slider__dot__scrubber" />
      </div>
    </template>

    <template #process="{ style }">
      <div
        class="vue-slider-process slider__process"
        :style="style"
      />
    </template>

    <template #tooltip="{ value }">
      <div class="slider__tooltip">
        {{ value }}
      </div>
    </template>
  </vue-slider>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'
import vueSlider from 'vue-slider-component'
import 'vue-slider-component/theme/default.css'

import ScrubberIcon from './assets/scrubber.svg?inline'

@Component({
  name: 'manipulator-double-slider',
  components: { ScrubberIcon, vueSlider }
})
export default class ManipulatorDoubleSlider extends Vue {
  @Prop({ required: true })
  value!: [number, number]

  @Prop({ default: 1 })
  min!: number

  @Prop({ default: 65535 })
  max!: number

  get sliderOptions () {
    return {
      eventType: 'auto',
      enableCross: false,
      width: '100%',
      height: 5,
      dotSize: 2,
      min: this.min,
      max: this.max,
      duration: 0,
      interval: 1,
      show: true,
      speed: 1,
      tooltipPlacement: 'bottom',
      railStyle: {
        background: this.$theme.getColor('colorAliceShadow'),
        backgroundPosition: 'center',
        height: '8px'
      },
      dotStyle: {},
      processStyle: {},
      tooltipStyle: {}
    }
  }

  get _value () {
    return this.value
  }

  set _value (val) {
    this.$emit('input', val)
    this.$emit('change', val)
  }
}
</script>

<style lang="scss" scoped>
.slider__dot {
  width: 2px;
  height: 36px;
  cursor: pointer;
  position: relative;
  background: $colorFeatherLight;
  margin-top: -3px;

  &:hover, &:active {
    box-shadow: 0px 5px 5px rgba(145, 169, 192, 0.3);
  }
}

.slider__dot--0 {
  .slider__dot__scrubber {
    color: $color90Black;
  }
}

.slider__dot--1 {
  .slider__dot__scrubber {
    color: $colorWhite;
  }
}

.slider__dot__scrubber {
  position: absolute;
  top: 0px;
  left: -4px;
  width: 9px;
  height: 36px;
  cursor: pointer;
}

:deep(.vue-slider-dot-tooltip) {
  cursor: pointer;
}

.slider__process {
  background: linear-gradient(90deg, #000000 0%, #FFFFFF 100%), #FFFFFF;
}

.slider__tooltip {
  @include typography(sm);
  @include row--center;
  background: rgba(0, 0, 0, 0.7);
  border-radius: 3px;
  color: #ffffff;
  text-align: center;
  padding: 5px 6px;
  margin-top: 25px;
  cursor: pointer;
}
</style>
