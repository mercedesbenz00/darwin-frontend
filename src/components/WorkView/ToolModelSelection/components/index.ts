export { default as ClassPill } from './ClassPill.vue'
export { default as ModelIcon } from './ModelIcon.vue'
