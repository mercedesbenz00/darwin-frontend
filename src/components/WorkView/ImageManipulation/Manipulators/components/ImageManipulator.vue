<template>
  <div class="image-manipulation__item">
    <div class="image-manipulation__item__header">
      <label class="image-manipulation__item__label">
        <slot name="label" />
      </label>
      <button
        v-if="optionalSync && $listeners['update:isSyncMode']"
        class="sync-mode-btn"
        :class="{ 'sync-mode-btn--active': isSyncMode }"
        @click="toggleSyncMode"
      >
        <chain-icon v-if="isSyncMode" />
        <unchain-icon v-else />
      </button>
      <button
        v-if="$listeners['refresh']"
        class="image-manipulation__item__button--reset"
        @click="$emit('refresh')"
      >
        <refresh-icon />
      </button>
      <label
        v-if="$slots['value']"
        class="image-manipulation__item__value"
      >
        <slot name="value" />
      </label>
    </div>

    <slot name="others" />

    <div
      v-if="$slots['slider']"
      class="image-manipulation__item__slider"
    >
      <slot name="slider" />
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'

import { ChainIcon, UnchainIcon } from '@/assets/icons/V1'

import ManipulatorSlider from './ManipulatorSlider.vue'
import RefreshIcon from './assets/refresh.svg?inline'

@Component({
  name: 'image-manipulator',
  components: {
    ManipulatorSlider,
    RefreshIcon,
    ChainIcon,
    UnchainIcon
  }
})
export default class ImageManipulator extends Vue {
  @Prop({ type: Boolean, default: false, required: false })
  isSyncMode!: boolean

  @Prop({ type: Boolean, default: false, required: false })
  optionalSync!: boolean

  toggleSyncMode (): void {
    this.$emit('update:isSyncMode', !this.isSyncMode)
  }
}
</script>

<style lang="scss" scoped>
.image-manipulation__item {
  @include col;
  align-items: center;
  width: 100%;
  &:not(:last-child) {
    margin-bottom: 20px;
  }
}

.image-manipulation__item__header {
  width: 100%;
  @include row;
  justify-content: center;
  margin-bottom: 10px;
}

.image-manipulation__item__label {
  flex: 1;
  width: 100%;
  @include typography(md-1, default);
  @include row;
  align-items: center;
  color: $colorSecondaryDark;
}

.image-manipulation__item__value {
  @include typography(md-1, default);
  min-width: 40px;
  text-align: right;
  color: $colorSecondaryLight;
}

.image-manipulation__item__button--reset {
  @include circle(22px);
  @include typography(md-1, headlines, 600);
  @include row--center;
  padding: 1px 0 0;
  color: transparent;
  background-color: transparent;
  margin: -2px 5px 0 0;
  padding: 5px;

  position: relative;

  &:hover {
    background-color: $colorSecondaryLight2;
  }

  img {
    width: 100%;
    height: 100%;
    object-fit: contain;
  }
}

.image-manipulation__item__slider {
  width: 100%;
}

.sync-mode-btn {
  @include circle(22px);
  padding: 5px;
  background-color: transparent;
  position: relative;

  &:hover {
    background-color: $colorSecondaryLight2;
  }

  & > svg {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    color: rgb(145, 169, 192);
    width: 15px;
    height: 15px;
    transition: fill .2s ease-in-out;
  }
}

.sync-mode-btn--active {
  & > svg {
    color: $colorBlue;
  }
}
</style>
