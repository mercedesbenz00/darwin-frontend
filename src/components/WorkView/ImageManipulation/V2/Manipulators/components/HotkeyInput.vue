<template>
  <div class="hotkey-input">
    <label class="hotkey-input__label">Hotkey</label>
    <div class="hotkey-input__box">
      <div
        class="hotkey-input__control hotkey-input__control--disabled"
      >
        <div
          v-tooltip="clickerSpecialTooltip"
          class="hotkey-input__clicker"
        >
          <span class="hotkey-special-text">Tab</span>
        </div>
      </div>

      <div
        class="hotkey-input__control"
        :class="{
          'hotkey-input__control--active': isAddingNumberHotkey
        }"
      >
        <div
          v-tooltip="clickerNumberTooltip"
          class="hotkey-input__clicker"
          @click="enableNumberHotkeyListener"
        >
          <more-horizontal-icon v-if="!numberHotKey || isAddingNumberHotkey" />
          <span v-else>{{ numberHotKey }}</span>
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator'

import { MoreHorizontalIcon } from '@/assets/icons/V1'
import { HotkeyString } from '@/components/WorkView/HotkeyInfo/types'

@Component({
  name: 'hotkey-input',
  components: { MoreHorizontalIcon }
})
export default class HotkeyInput extends Vue {
  @Prop({ required: false, default: null })
  hotkey!: HotkeyString | null

  isAddingNumberHotkey: boolean = false

  get speicalHotKey (): string | null {
    if (this.hotkey && this.hotkey.keys && this.hotkey.keys.length) {
      return this.hotkey.keys[0]
    }
    return null
  }

  get numberHotKey (): string | null {
    if (this.hotkey && this.hotkey.keys && this.hotkey.keys.length > 1) {
      return this.hotkey.keys[1]
    }
    return null
  }

  get clickerNumberTooltip (): { content: string } {
    return {
      content: 'Press a key between 1 and 9 to select this'
    }
  }

  get clickerSpecialTooltip (): { content: string } {
    return {
      content: 'You can not input other special key'
    }
  }

  beforeDestroy (): void {
    this.disableNumberHotkeyListener()
  }

  enableNumberHotkeyListener (): void {
    this.isAddingNumberHotkey = true
    document.addEventListener('keydown', this.onNumberKeydown)
  }

  disableNumberHotkeyListener (): void {
    this.isAddingNumberHotkey = false
    document.removeEventListener('keydown', this.onNumberKeydown)
  }

  onNumberKeydown (event: KeyboardEvent): void {
    event.preventDefault()
    event.stopPropagation()

    if (event.key === 'Escape') {
      this.disableNumberHotkeyListener()
      return
    }

    // Instead of keyCode which is deprecated
    if (!['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'].includes(event.key)) {
      this.$store.dispatch('toast/notify'
        , { content: 'Type a number between 1 and 9 to set a hotkey' })
      return
    }

    const newHotkey: HotkeyString = this.hotkey ? { ...this.hotkey } : { keys: [] }

    if (newHotkey.keys.length > 1) {
      newHotkey.keys[1] = event.key
    } else {
      newHotkey.keys = [newHotkey.keys.length ? newHotkey.keys[0] : '', event.key]
    }
    this.$emit('update:hotkey', newHotkey)
    this.disableNumberHotkeyListener()
  }
}
</script>

<style lang="scss" scoped>
.hotkey-input {
  align-items: center;
  width:45px;
  margin-top: -1.4rem;
  flex-shrink: 0;
}

.hotkey-input__box {
  display: -webkit-box;
  gap: 8px;
}

.hotkey-input__control {
  position: relative;
  width: 39px;
  height: 36px;
  background: $colorAliceBlue;
  border: 2px solid $colorAliceShadow;
  border-radius: $border-radius-default;
  margin-top: 0.5rem;
}

.hotkey-input__clicker {
  @include row--center;
  position: absolute;
  top: -2px;
  left: -2px;
  width: 39px;
  height: 36px;
  cursor: pointer;
  background: transparent;
  box-shadow: inset 0px 2px 4px rgba(255, 255, 255, 0.3),
    inset 0px -2px 2px rgba(145, 169, 192, 0.4);
  border-radius: $border-radius-default;
  transition: all .1s;

  svg {
    max-width: 70%;
    color: $colorAliceNight;
  }

  span {
    @include typography(md, default, bold);
    @include noSelect;
    color: $color90Black;
  }

  span.hotkey-special-text {
    @include typography(md, default, bold);
  }

  &:hover {
    background: $colorAliceShade;
  }

  &:active {
    background: $colorAliceShadow;
  }
}

.hotkey-input__control--active {
  background: $colorAliceShadow;

  .hotkey-input__clicker:hover {
    background: $colorAliceShadow;
  }
}

.hotkey-input__control--disabled {
  opacity: 0.8;

  .hotkey-input__clicker:hover,
  .hotkey-input__clicker:active {
    background: transparent;
    cursor: not-allowed;
  }
}

.hotkey-input__label {
  @include typography(sm, default, bold);
  white-space: nowrap;
  color: $colorAliceNight;
  @include noSelect;
}
</style>
