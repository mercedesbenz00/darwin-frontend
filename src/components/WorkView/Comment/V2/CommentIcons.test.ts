import { createLocalVue, shallowMount } from '@vue/test-utils'
import { Ref, ref } from 'vue'
import Vuex from 'vuex'

import { createMockTheme } from 'test/unit/components/mocks'
import { createEditorV2 } from 'test/unit/createEditorV2'
import createTestStore from 'test/unit/createTestStore'
import {
  buildDatasetPayload,
  buildTeamPayload,
  buildV2CommentThreadPayload,
  buildV2DatasetItemPayload
} from 'test/unit/factories'

import { payloadToEditorThread } from '@/engineV2/commentHelpers'
import { Editor } from '@/engineV2/editor'

import CommentIcons from './CommentIcons.vue'

jest.mock('@/engineV2/workers/FramesLoaderWorker')
jest.mock('uuid', () => ({ v4: (): string => 'fake-view-uuid' }))

const localVue = createLocalVue()
localVue.use(Vuex)

let store: ReturnType<typeof createTestStore>
let provide: { editorV2: Ref<Editor> }
let propsData: { viewId: string }
const mocks = { $theme: createMockTheme() }
const item = buildV2DatasetItemPayload({ id: 'item-id' })

const initEditor = (store: ReturnType<typeof createTestStore>): Ref<Editor> => {
  const editor: Editor = createEditorV2(
    store,
    { type: 'simple', views: [{ item, file: item.slots[0] }] }
  )
  // type inference here is acting weird, so we need to cast
  return ref(editor) as Ref<Editor>
}

beforeEach(() => {
  store = createTestStore()
  store.commit('team/SET_TEAMS', [buildTeamPayload({ id: 1, slug: 'v7' })])
  store.commit('workview/SET_DATASET', buildDatasetPayload({ id: 10, team_id: 1, team_slug: 'v7' }))

  // handles loading of frames
  jest.spyOn(store, 'dispatch').mockResolvedValue({
    data: { slot_sections: [] }
  })

  provide = { editorV2: initEditor(store) }
  propsData = { viewId: provide.editorV2.value!.activeView.id }
})

describe('when no threads', () => {
  it('renders', () => {
    const wrapper = shallowMount(CommentIcons, { localVue, mocks, propsData, store, provide })
    expect(wrapper.find('.comment-icons').exists()).toBe(true)
  })

  it('matches snapshot', () => {
    const wrapper = shallowMount(CommentIcons, { localVue, mocks, propsData, store, provide })
    expect(wrapper).toMatchSnapshot()
  })
})

describe('when some threads', () => {
  beforeEach(() => {
    const thread1 = buildV2CommentThreadPayload({ id: 'thread-1-id', section_index: 0 })
    const thread2 = buildV2CommentThreadPayload({ id: 'thread-2-id', section_index: 0 })

    provide.editorV2.value.activeView.commentManager.setCommentThreads(
      [thread1, thread2].map(payloadToEditorThread)
    )
  })

  it('matches snapshot', () => {
    const wrapper = shallowMount(CommentIcons, { localVue, mocks, propsData, store, provide })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders thread icons', () => {
    const wrapper = shallowMount(CommentIcons, { localVue, mocks, propsData, store, provide })
    expect(wrapper.findAll('comment-thread-icon-stub').length).toBe(2)
  })
})
