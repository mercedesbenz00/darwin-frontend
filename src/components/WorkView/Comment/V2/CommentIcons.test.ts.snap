// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when no threads matches snapshot 1`] = `<div class="comment-icons"></div>`;

exports[`when some threads matches snapshot 1`] = `
<div class="comment-icons">
  <comment-thread-icon-stub threadid="thread-1-id" viewid="fake-view-uuid"></comment-thread-icon-stub>
  <comment-thread-icon-stub threadid="thread-2-id" viewid="fake-view-uuid"></comment-thread-icon-stub>
</div>
`;
